src/vivarium_nih_us_cvd/components/treatment.py

```diff
@@ -429,15 +430,154 @@ def apply_sbp_treatment_ramp(
+    def apply_ldlc_treatment_ramp(
+        self,
+        pop_visitors: pd.DataFrame,
+        ldlc_pipeline: Optional[Pipeline] = None,
+        sbp_pipeline: Optional[Pipeline] = None,
+    ) -> pd.DataFrame:
+        """Applies the LDL-C treatment ramp
+
+        Arguments:
+            pop_visitors: dataframe subset to simulants visiting the doctor
+            ldlc_pipeline: the ldl-c exposure pipeline to use when calculating
+                measured ldl-c values; defaults to the values adjusted for
+                population treatment effects in gbd data except during
+                initialization
+            sbp_pipeline: the sbp exposure pipeline to use when calculating
+                ASCVD; defaults to the values adjusted for population treatment
+                effects in gbd data except during initialization
+        """
+        if not ldlc_pipeline:
+            ldlc_pipeline = self.ldlc
+        if not sbp_pipeline:
+            sbp_pipeline = self.sbp
+        overcome_therapeutic_inertia = pop_visitors[
+            self.randomness.get_draw(
+                pop_visitors.index,
+                additional_key="ldlc_therapeutic_inertia",
+            )
+            > data_values.LDLC_THERAPEUTIC_INERTIA_NO_START
+        ].index
+        currently_medicated = pop_visitors[
+            pop_visitors[data_values.COLUMNS.LDLC_MEDICATION]
+            != data_values.LDLC_MEDICATION_LEVEL.NO_TREATMENT.DESCRIPTION
+        ].index
+        not_currently_medicated = pop_visitors.index.difference(currently_medicated)
+
+        ascvd = self.get_ascvd(pop_visitors=pop_visitors, sbp_pipeline=sbp_pipeline)
+        measured_ldlc = self.get_measured_ldlc(
+            index=pop_visitors.index,
+            mean=data_values.MEASUREMENT_ERROR_MEAN_LDLC,
+            sd=data_values.MEASUREMENT_ERROR_SD_LDLC,
+            exposure_pipeline=ldlc_pipeline,
+        )
+
+        # Generate other useful helper indexes
+        elevated_ascvd = ascvd[ascvd >= data_values.ASCVD_THRESHOLD.LOW].index
+        high_ascvd = ascvd[ascvd >= data_values.ASCVD_THRESHOLD.HIGH].index
+        mid_ascvd = elevated_ascvd.difference(high_ascvd)
+        elevated_ldlc = measured_ldlc[measured_ldlc >= data_values.LDLC_THRESHOLD.LOW].index
+        high_ldlc = measured_ldlc[measured_ldlc >= data_values.LDLC_THRESHOLD.HIGH].index
+        mid_ldlc = elevated_ldlc.difference(high_ldlc)
+        mask_history_mi = (
+            pop_visitors[models.MYOCARDIAL_INFARCTION_MODEL_NAME]
+            != models.MYOCARDIAL_INFARCTION_SUSCEPTIBLE_STATE_NAME
+        )
+        mask_history_is = (
+            pop_visitors[models.ISCHEMIC_STROKE_MODEL_NAME]
+            != models.ISCHEMIC_STROKE_SUSCEPTIBLE_STATE_NAME
+        )
+        history_mi_or_is = pop_visitors[mask_history_mi | mask_history_is].index
+        no_history_mi_or_is = pop_visitors.index.difference(history_mi_or_is)
+
+        # [Treatment ramp ID E] Simulants who overcome therapeutic inertia, have
```

Comment:
lol I'd actually done this already but didn't love the reference to previous indexes. BUT, happy to change it back if it's more readable to others (ie you); especially since I'd literally done that initially